### PR TITLE
Split math code into multiple files

### PR DESCRIPTION
**Pull request by CodingDive**
Now that we have made some nice bug fixes and visual improvements with the math editor, I just took a few components/render fns and extracted them into their own files/components. 

Also put the DOM element selection into an effect for concurrent mode :tada:

### CODE REVIEW

**Review by hejtful — APPROVED** [review]
Nice!

File: src/serlo-editor/math/editor.tsx [hunk 1]
```diff
@@ -104,85 +53,30 @@ export function MathEditor(props: MathEditorProps) {
     }
   )
 
-  const isVisualMode = visual && !hasError
+  useEffect(() => {
+    if (typeof window !== 'undefined') {
```
Comment (elbotho) [inline src/serlo-editor/math/editor.tsx]: useEffect only runs on client so the check is not needed any more right?

File: src/serlo-editor/math/editor.tsx [hunk 2]
```diff
@@ -262,7 +156,7 @@ export function MathEditor(props: MathEditorProps) {
             )}
             {isVisualMode && (
               <button
-                onMouseDown={() => setHelpOpen(true)}
+                onMouseDown={() => void setIsHelpOpen(true)}
```
Comment (elbotho) [inline src/serlo-editor/math/editor.tsx:157]: why do you prefer this?

File: src/serlo-editor/math/editor.tsx [hunk 3]
```diff
@@ -271,43 +165,52 @@ export function MathEditor(props: MathEditorProps) {
...
-            aria-label="Close math formula editor"
-            data-qa="plugin-math-close-formula-editor"
-          >
-            <FaIcon icon={faXmark} />
-          </button>
-        </div>
-        {!isVisualMode && (
-          <MathEditorTextArea {...props} defaultValue={state} />
-        )}
+function MathEditorOverlayPortal({
```
Comment (elbotho) [inline src/serlo-editor/math/editor.tsx]: Why is this called `…OverlayPortal`? 
Also I think I'd prefer to have this in an own file now.

File: src/serlo-editor/math/math-help-modal.tsx [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+export const MathHelpModal = ({
+  isHelpOpen,
+  setIsHelpOpen,
+}: {
+  isHelpOpen: boolean
+  setIsHelpOpen: (isOpen: boolean) => void
+}) => {
+  const mathStrings = useEditorStrings().plugins.text.math
+  return (
+    <Modal
```
Comment (elbotho) [inline src/serlo-editor/math/math-help-modal.tsx]: maybe follow up: using `ModalWithCloseButton` here would probably be a lot simpler.

**Review by elbotho — CHANGES_REQUESTED** [review]

File: src/serlo-editor/math/editor.tsx (hunk 1 above)
Comment (CodingDive) [reply src/serlo-editor/math/editor.tsx]: No, that's useLayoutEffect. useEffect works on the server too

File: src/serlo-editor/math/editor.tsx (hunk 2 above)
Comment (CodingDive) [reply src/serlo-editor/math/editor.tsx:157]: I just like to prefix booleans with "is", "has", etc. The `void` I like to symbolize that there is no return value.

File: src/serlo-editor/math/math-help-modal.tsx (hunk 4 above)
Comment (CodingDive) [reply src/serlo-editor/math/math-help-modal.tsx]: Oh, cool! Will take a look

File: src/serlo-editor/math/editor.tsx (hunk 1 above)
Comment (elbotho) [reply src/serlo-editor/math/editor.tsx]: > Effects only run on the client. They don’t run during server rendering.

https://react.dev/reference/react/useEffect

**Review by elbotho — APPROVED** [review]
thanks for the improvement!

**Pull request merged**